JSDemos/Demos/TreeList/RowEditing/jQuery/index.js

```diff
@@ -10,7 +14,9 @@ $(() => {
       mode: 'row',
       allowUpdating: true,
       allowDeleting: true,
-      allowAdding: true,
+      allowAdding: (args)=>{
+        return isDeleteAllowed(args.row.data)
+      },
```

Comment:
```suggestion
      allowDeleting: (args)=>{
        return isDeleteAllowed(args.row.data)
      },
```

Also let's not extract it to separate function, it's too small for it

Reply:
Fixed in a separate commit